dali/operators/decoder/inflate/inflate_gpu.cc

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <nvcomp/lz4.h>
+
+#include <memory>
+#include <vector>
+
+#include "dali/core/backend_tags.h"
+#include "dali/core/common.h"
+#include "dali/core/static_switch.h"
+#include "dali/core/tensor_shape.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/operators/decoder/inflate/inflate.h"
+#include "dali/operators/decoder/inflate/inflate_gpu.h"
+#include "dali/operators/decoder/inflate/inflate_params.h"
+#include "dali/pipeline/data/sequence_utils.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+
+namespace inflate {
+
+template <typename Status>
+void nvcompCall(Status status) {
+  if (status != nvcompSuccess) {
+    throw std::runtime_error(
+        make_string("Inflate GPU op error: nvComp returned non-zero status: `", status, "`."));
+  }
+}
+
+class InflateOpGpuLZ4Impl : public InflateOpImplBase<GPUBackend> {
+ public:
+  explicit InflateOpGpuLZ4Impl(const OpSpec &spec) : InflateOpImplBase<GPUBackend>{spec} {}
+
+  void RunImpl(Workspace &ws) override {
+    const auto &input = ws.template Input<GPUBackend>(0);
+    auto &output = ws.template Output<GPUBackend>(0);
+    auto total_chunks_num = params_.GetTotalChunksNum();
+    output.SetLayout(params_.GetOutputLayout());
+    SetupInChunks(input);
+    SetupOutChunks(output);
+    auto stream = ws.stream();
+
+    kernels::DynamicScratchpad scratchpad({}, stream);
+    size_t *actual_out_sizes = scratchpad.AllocateGPU<size_t>(total_chunks_num);
+
+    size_t *in_sizes;
+    void const **in;
+    size_t *out_sizes;
+    void **out;
+    std::tie(in_sizes, in, out_sizes, out) = scratchpad.ToContiguousGPU(
```

Comment:
Hmmm.... would that work?
```suggestion
    auto [in_sizes, in, out_sizes, out] = scratchpad.ToContiguousGPU(
```

Reply:
I wasn't sure if it end up as .cu or .cc file, but now, yeah, sure.